### PR TITLE
fix statement containing columnsegment

### LINKED ISSUE #4671: Parsing more segments
#### TODO
- RenameColumnSegment
    - [x] Instead of `columnName`, parse `ColumnNameSegment` as its member.

- ColumnAfterPositionSegment
    - [x] Instead of `columnName`, parse `ColumnNameSegment` as its member.

- ColumnPositionSegment
    - [x] Instead of `columnName`, parse `ColumnNameSegment` as its member.

- ColumnDefinitionSegment
    - [x] Instead of `columnName`, parse `ColumnNameSegment` as its member.

- ColumnDefinitionSegment
    - [x] Instead of `columnName`, parse `ColumnNameSegment` as its member.
    - [x] Instead of `dataType`, parse `DataTypeSegment` as its member.

### Notes
- Refer to `TableNameSegment` to create `ColumnNameSegment`.
- `ComparisonOperatorType`  is expected to be `enum`.

### PR DESCRIPTION
**Pull request by jingshanglu**
ref #4671.

Changes proposed in this pull request:
- Instead of columnName, parse ColumnSegment as  member of RenameColumnSegment
- Instead of columnName, parse ColumnSegment as  member of ColumnAfterPositionSegment
- Instead of columnName, parse ColumnSegment as  member of ColumnPositionSegment
- Instead of columnName, parse ColumnSegment as  member of ColumnDefinitionSegment

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/segment/ddl/column/position/ColumnAfterPositionSegment.java [hunk 1]
```diff
@@ -27,7 +28,7 @@
     
     private final String afterColumnName;
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/segment/ddl/column/position/ColumnAfterPositionSegment.java]: Why do not change it as ColumnSegment?

Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/segment/ddl/column/position/ColumnAfterPositionSegment.java]: Why do not you change it as `columnSegment`?

Comment (jingshanglu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/segment/ddl/column/position/ColumnAfterPositionSegment.java]: ColumnPositionSegment include columnName, and the type of columnName is ColumnSegment

Comment (jingshanglu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/segment/ddl/column/position/ColumnAfterPositionSegment.java]: the same as above

Comment (jingshanglu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/segment/ddl/column/position/ColumnAfterPositionSegment.java]: ColumnPositionSegment include columnName, the type od columnName is ColumnSegment.

**Pull request merged**